Implementation of Topic List Page for each Folder ID

takahashi-akari opened the pull request:
This pull request introduces the implementation of pages for each Folder ID, essentially creating a Topic List Page.

The changes included are as follows:

- Integrated data retrieval from two APIs necessary for the functionality of the Topic List Page.
- Implemented pagination feature to enhance user navigation experience on the Topic List Page.
- Incorporated a table view for displaying the topic list, improving readability and data organization.
- Ensured smartphone compatibility for improved accessibility and optimal user experience across different devices.

Please review the changes and let me know if any adjustments are needed.

FinnRG commented on VocaDbWeb/New/pages/discussion/folders/[id].tsx:
```diff
@@ -1,7 +1,153 @@
+		),
+	]);
+
+	const [folders, topics] = await Promise.all([
```
This can be typed using `DiscussionFolderContract[]` and `DiscussionTopicContract[]`:
`const [folders, topics]: [DiscussionFolderContract[], DiscussionTopicContract[]] =`

FinnRG commented on VocaDbWeb/New/pages/discussion/folders/[id].tsx:
```diff
@@ -1,7 +1,153 @@
+		apiResp[1].json().then((topics) => topics.items),
+	]);
+
+	const folder = folders.find((folder: any) => folder.id === id);
```
The typings should make the `any` unnecessary.

FinnRG commented on VocaDbWeb/New/pages/discussion/folders/[id].tsx:
```diff
@@ -1,7 +1,153 @@
+		return { notFound: true };
+	}
+
+	const totalPage = Math.ceil(parseFloat(folder.topicCount) / perPage);
```
`folder.topicCount` is a number and doesn't require ``parseFloat`

FinnRG commented on VocaDbWeb/New/pages/discussion/folders/[id].styles.ts:
```diff
@@ -0,0 +1,165 @@
+    folderTable: {
+        width: '100%',
+		margin: '0 auto',
+		maxWidth: '1024px',
```
This could either get increased or removed

FinnRG reviewed (changes requested):
Thank you for your PR. The page already looks really good!

takahashi-akari commented on VocaDbWeb/New/pages/discussion/folders/[id].tsx:
```diff
@@ -1,7 +1,153 @@
+							<td>{topic.commentCount}</td>
+							<td>{new Date(topic.created).toLocaleString()}</td>
+							<td>
+								{topic.comments.length > 0
+									? new Date(
+											topic.comments[topic.comments.length - 1]?.created
+									  ).toLocaleString()
+									: ''}
```
It was difficult to move into a variable, so I simplified it.

FinnRG commented on VocaDbWeb/New/pages/discussion/folders/[id].styles.ts:
```diff
@@ -0,0 +1,162 @@
+			},
+		},
+
+		'& td': {
+			width: '100%',
+			wordBreak: 'break-word',
```
I think you can put the duplicate code inside a `& td, & th` clause

FinnRG commented on VocaDbWeb/New/pages/discussion/folders/[id].tsx:
```diff
@@ -1,7 +1,148 @@
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+	const styles = useStyles();
+	const setPage = (page: number) => {
+		window.location.href = `/discussion/folders/${folder.id}?page=${page}`;
```
Please use the nextjs [useRouter ](https://nextjs.org/docs/pages/api-reference/functions/use-router) api

takahashi-akari commented:
Thank you for your review. 
Based on the review of PR #1546, I have made some modifications. 
Could you please review it again? 

Here's what's changed:
- Implemented the Title component
- Utilized nth-of-type and first-of-type
- Consolidated duplicate styling
- Implemented useRouter for page transitions in pagination

FinnRG commented:
Thank you for your contributions!
